doc/users/explain/figures.rst

```diff
@@ -0,0 +1,188 @@
+
+.. _figure_explanation:
+
+================================================
+Creating, viewing, and saving Matplotlib Figures
+================================================
+
+.. plot::
+    :include-source:
+
+    fig = plt.figure(figsize=(2, 2), facecolor='0.8', layout='constrained')
+    ax = fig.add_subplot()
```

Comment:
Maybe add first
```
    fig = plt.figure(figsize=(2, 2), facecolor='0.8')
    fig.text(0.1, 0.5, "This is a figure")
```
to illustrate the basic conecpt of adding content with out the need to already dive into the `Axes` / subplot complexity.

And then follow up with the Axes discussion.

Reply:
I'm not in favour of adding a relatively obscure artist to the first example - an Axes is what 99.99% of visualizations start with.  I personally wasn't even aware that Figure has a text method, so it's pretty obscure.

Reply:
Though this inspires a new section below on adding artists....

Reply:
I'm trying to make a minimal illustration of a figure. Maybe even
```
plt.figure(facecolor='lightblue')
```
is enough.

Side-remark: Please don't use the somewhat obscure '0.8' greyscale notation in a basic example. That is unnecessarily confusing.

If the empty figure is too little, you could use `suptitle()`, which is simpler than `text()`:

```
plt.figure(facecolor='lightblue')
plt.suptitle('This is a figure')
```

Reply:
I want a minimal typical example of a figure in the context of what the reader has likely seen before, not an artificial example with nothing on it.  Readers are not starting with zero context here.  

I can easily change the background colour.

Reply:
Updated to change the color and added some annotation on the Axes.

Reply:
> I want a minimal typical example of a figure in the context of what the reader has likely seen before, not an artificial example with nothing on it.

I would agree for a tutorial. In explanations however, I think we should clearly isolate the figure. Both visually, and from the code complexity

```
fig = plt.figure(facecolor='lightblue')
fig.suptitle('This is a figure')
```
![grafik](https://user-images.githubusercontent.com/2836374/213865657-c9cc31d0-7888-4e65-b309-a62dbf637905.png)

is much clearer than

```
fig = plt.figure(figsize=(2, 2), facecolor='lightskyblue',
                 layout='constrained')
fig.suptitle('Figure')
ax = fig.add_subplot()
ax.set_title('Axes', loc='left', fontstyle='oblique', fontsize='medium')
```

![grafik](https://user-images.githubusercontent.com/2836374/213865603-5a54c563-e3a5-4783-838f-c338589141e3.png)



> Readers are not starting with zero context here.

Reply:
Again, I think this initial example is more effective in the context of something the viewer would have seen - 99.9999% of figures have an Axes.  The complexity is all in separating the two objects in code so that the reader can understand how Matplotlib differentiates them internally.  I don't feel that an empty figure is a good use of space or the reader's time.

Reply:
I beg to differ on the overall statement. 

> in the context of something the viewer would have seen - 99.9999% of figures have an Axes.

Note also that `plt.figure(); fig.add_subplot()` is rarely used (`plt.subplots()` being far more common and IMHO recommended). There is value in using the figure() call here explicitly and I'm not suggesting to change, but this fact weakens the familiarity argument. The generated figure is familiar, but the associated code is not.

I have the feeling that the current approach is neither fish nor fowl. Maybe, if you wan't to start with the familiar, really do subplots() and then explain that this does the two statements under the hood.

I really appreciate your work on the docs, and I'd like to make sure they are of maximal use for a diverse audience. I would like to encourage  to carefully consider how the descriptions work for users with different backgrounds. We as authors have a very good understanding, but throwing in variations (add_subplot vs subplots) or concepts (In the text Artists are mentioned without explanation or even only a link for that) can be obstacles for some readers.

Unfortunately, I don't have much time currently. So I will withdraw from the PR and the discussion.